src/crimson/os/seastore/segment_cleaner.h

```diff
@@ -44,6 +95,249 @@ struct segment_info_t {
   bool is_open() const {
     return state == Segment::segment_state_t::OPEN;
   }
+  void set_unavail() {
+    usii = smit->usis.emplace(smit->usis.end(), this);
+  }
+};
+
+class SegmentCleaner;
+
+// for keeping track of segment managers' various information,
+// like empty segments, opened segments and so on.
+template <typename SegmentInfoT>
+class segment_info_set_t {
```

Comment:
Other than the unavailable segment thing, segment_info_set_t makes sense to me now.